plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1023,7 +1030,7 @@ public void rollback()
         TupleDomain<IcebergColumnHandle> newUnenforcedConstraint = remainingConstraint
                 // TODO: Remove after completing https://github.com/trinodb/trino/issues/8759
                 // Only applies to the unenforced constraint because structural types cannot be partition keys
-                .filter((columnHandle, predicate) -> !isStructuralType(columnHandle.getType()))
+                .filter((columnHandle, predicate) -> !isStructuralType(columnHandle.getType()) && !columnHandle.isPathColumn())
```

Comment:
Pls extract the filter logic either to a variable function or a private function and explain there in the comments what each of the filter is used for.
The reader of the code may think now that also your newly added condition is linked to #8759